bundler/lib/bundler/injector.rb

```diff
@@ -269,5 +270,13 @@ def cross_check_for_errors(gemfile_path, original_deps, removed_deps, initial_ge
     def show_warning(message)
       Bundler.ui.info Bundler.ui.add_color(message, :yellow)
     end
+
+    def convert_autorequire(autorequire)
+      require 'json'
+      autorequire
+        .yield_self { |r| (r.one? ? r.first : r) }   # Unwrap array
+        .yield_self { |r| r == "false" ? false : r } # Convert false to boolean
```

Comment:
`yield_self` has been [introduced in ruby 2.5 ](https://www.ruby-lang.org/en/news/2017/12/25/ruby-2-5-0-released/) and [bundler currently supports ruby 2.3 and up](https://github.com/rubygems/rubygems/issues/3260). So I think it is best to change the code to support ruby 2.3.